src/convertToObject.js

```diff
@@ -7,6 +7,19 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  return sourceString
+    .split(';')
+    .map((entry) => entry.trim())
+    .filter((entry) => entry)
+    .reduce((cssProperties, entry) => {
+      const [key, value] = entry
+        .split(':')
+        .map((spltEntry) => spltEntry.trim());
+
+      cssProperties[key] = value;
+
+      return cssProperties;
+    }, {});
```

Comment:
It's good to see that you've used a combination of `split`, `map`, `filter`, and `reduce` to process the string and convert it into an object. This follows functional programming principles and avoids unnecessary mutations.